lib/Dialect/HW/HWInstanceImplementation.cpp

```diff
@@ -256,9 +260,24 @@ LogicalResult instance_like_impl::verifyInstanceOfHWModule(
     return failure();
 
   if (parameters) {
+    SmallVector<Type> rawModParameters;
+    SmallVector<Type> resolvedModParameters;
+
+    auto modParameters = module->getAttrOfType<ArrayAttr>("parameters");
+    for (auto param : modParameters) {
+      auto paramDecl = cast<ParamDeclAttr>(param);
+      rawModParameters.push_back(paramDecl.getType());
+    }
```

Comment:
super nit: Please reserve results and use `getAsRange<..>`.
```suggestion
    auto modParameters = module->getAttrOfType<ArrayAttr>("parameters");
    SmallVector<Type> rawModParameters, resolvedModParameters;
    rawModParameters.reserve(modParameters.size());
    resolvedModParameters.reserve(modParameters.size());
    for (auto paramDecl : modParameters.getAsRange<ParamDeclAttr>())
      rawModParameters.push_back(paramDecl.getType());
```